packages/starknet-snap/src/index.ts

```diff
@@ -208,3 +211,86 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
       throw new Error('Method not found.');
   }
 };
+
+export const onInstall: OnInstallHandler = async () => {
+  const component = panel([
+    text("Your MetaMask wallet is now compatible with Starknet!"),
+    text(`To manage your Starknet account and send and receive funds, visit the [companion dapp for Starknet](${dappUrl(process.env.SNAP_ENV)}).`),
+  ]);
+
+  await snap.request({
+    method: 'snap_dialog',
+    params: {
+      type: 'alert',
+      content: component,
+    },
+  });
+};
+
+export const onUpdate: OnUpdateHandler = async () => {
+  const component = panel([
+    text("Features released with this update:"),
+    text(
+      "**Home Page**: You can now see your account and balance on the Snap homepage.",
+    ),
+    text(
+      "To get to the Starknet Snap homepage, from the MetaMask menu, click **Snaps**, then click on the Starknet Snap."
+    ),
+  ]);
+
+  await snap.request({
+    method: 'snap_dialog',
+    params: {
+      type: 'alert',
+      content: component,
+    },
+  });
+};
+
+export const onHomePage: OnHomePageHandler = async () => {
+  const panelItems = [];
+  try {
+    const state: SnapState = await snap.request({
+      method: 'snap_manageState',
+      params: {
+        operation: 'get',
+      },
+    });
+
+    if (state && state.accContracts.length > 0) {
```

Comment:
A account doesnt exist in state, it doesnt means it has not deployed

we should recover the account, and check if the account has deployed by chain id if you are looking for a checking where a account has deployed or not

else, please modify the message on else case

Reply:
Done. Updated the `else` message.